src/rwlock.rs

```diff
@@ -381,16 +317,14 @@ impl<T: Default + ?Sized> Default for RwLock<T> {
 
 /// The future returned by [`RwLock::read`].
 pub struct Read<'a, T: ?Sized> {
-    /// The lock that is being acquired.
-    lock: &'a RwLock<T>,
-
-    /// The last-observed state of the lock.
-    state: usize,
+    raw: RawRead<'a>,
 
-    /// The listener for the "no writers" event.
-    listener: Option<EventListener>,
+    value: *const T,
```

Comment:
Add comments to these fields, especially this one explaining why we're doing it this way.